From 1ac97a7087e9f7920e73b7b2bb4871930cb081df Mon Sep 17 00:00:00 2001
From: Stefan Mayer <stevijo.mayer@googlemail.com>
Date: Sat, 31 May 2025 21:20:48 +0200
Subject: [PATCH 2/2] fix ike packages not forwarded to linux

---
 vnet/ipsec/ipsec_input.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/vnet/ipsec/ipsec_input.c b/src/vnet/ipsec/ipsec_input.c
index 6a25f6c58..139e7dbac 100644
--- a/vnet/ipsec/ipsec_input.c
+++ b/vnet/ipsec/ipsec_input.c
@@ -731,8 +731,7 @@ VLIB_NODE_FN (ipsec4_input_node) (vlib_main_t * vm,
 	   * UDP-encapsulated ESP and IKE."
 	   * RFC5996 Section 3.1: "IKE messages use UDP ports 500 and/or 4500"
 	   */
-	  if ((clib_host_to_net_u16 (500) == udp0->dst_port) ||
-	      (clib_host_to_net_u16 (4500) == udp0->dst_port))
+	  if (clib_host_to_net_u16 (4500) == udp0->dst_port)
 	  {
 	    esp0 = (esp_header_t *) ((u8 *) udp0 + sizeof (udp_header_t));
 
-- 
2.44.1

